### ISSUE TITLE
精度问题

### ISSUE THREAD

**Issue by VikingDeng** [CVHub520/X-AnyLabeling#721]
### Search before asking

- [X] I have searched the X-AnyLabeling [issues](https://github.com/CVHub520/X-AnyLabeling/issues) and found no similar bug report.

### X-AnyLabeling Component

Model

### Bug

你好，我正在使用自己训练的yolo11s模型进行标注任务，但是我发现对于同一个数据，在x-anyLabeling上模型的预测结果比我直接用模型预测要好得多，具体来说
这是x-anylabeling的预测结果，置信度都很高而且准确，最低0.7，其他是0.8 0.9
![image](https://github.com/user-attachments/assets/ad796298-4ede-4a24-8875-180467e72719)
但是这是我自己用程序推理的结果，置信度很低，最高0.7，其他是0.5 0.3
![image](https://github.com/user-attachments/assets/f18182fb-9e28-40a4-bbff-ed447abaa0c2)
但最重要的是，我对比过两者输出的bbox，他们最大的差别在于置信度，而bbox的位置基本都是对应的
为什么？x-anylabeling做了什么额外的处理以至于对于同一个模型同一个样本他的推理结果要比我自己程序的结果好的多吗
这是我的单个样本推理的程序(我曾怀疑过是模型数据格式的问题，但事实上onnx和pt文件推理结果是一样的)

```python
import torch
import cv2
from pathlib import Path
import json
from ultralytics import YOLO

def process_single_image(model_path, image_path, input_size=640, conf_threshold=0.3, iou_threshold=0.45):
    try:
        model = YOLO(model_path)
        img = cv2.imread(image_path)
        if img is None:
            raise ValueError(f"Could not read image {image_path}")
        img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
        original_height, original_width = img.shape[:2]
        img_resized = cv2.resize(img, (input_size, input_size), interpolation=cv2.INTER_LINEAR)
        img_tensor = torch.from_numpy(img_resized).permute(2, 0, 1).float() / 255.0
        results = model.predict(source=img_tensor.unsqueeze(0), imgsz=input_size, conf=conf_threshold, iou=iou_threshold, verbose=False)
        label_data = []
        image_id = Path(image_path).stem

        has_low_confidence = False
        for box in results[0].boxes:
            coords = box.xyxy.cpu().numpy().flatten()
            x1, y1, x2, y2 = map(float, coords)
            confidence = float(box.conf.item())

            if confidence < conf_threshold:
                has_low_confidence = True
            x1 = x1 * (original_width / input_size)
            y1 = y1 * (original_height / input_size)
            x2 = x2 * (original_width / input_size)
            y2 = y2 * (original_height / input_size)
            x1, y1, x2, y2 = round(x1, 2), round(y1, 2), round(x2, 2), round(y2, 2)
            confidence = round(confidence, 2)

            label_data.append({
                "coordi": [x1, y1, x2, y2],
                "confi": confidence
            })

        print(f"Prediction for image: {image_id}")
        print(json.dumps(label_data, indent=2))

        if has_low_confidence:
            print(f"Low confidence detected for image {image_id}.")
            
    except Exception as e:
        print(f"Error processing image {image_path}: {e}")


if __name__ == "__main__":
    model_path = "model/yolo11s-2500.pt"
    image_path = "../images/1265030679.png"
    process_single_image(model_path, image_path, conf_threshold=0.3, iou_threshold=0.45)

```

### Information

我觉得应该无关

### Link to a Reproducible Demonstration Video

也无关

### Execution Mode

Source Code

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by VikingDeng** [comment]
以防万一我再给出更多的信息
这是x-anylabeling用我的模型推理出的bbox的信息
{
  "version": "2.4.4",
  "flags": {},
  "shapes": [
    {
      "label": "figure",
      "score": 0.9150258302688599,
      "points": [
        [
          29.914649963378906,
          428.7963348388672
        ],
        [
          116.09727401733399,
          428.7963348388672
        ],
        [
          116.09727401733399,
          510.57238006591797
        ],
        [
          29.914649963378906,
          510.57238006591797
        ]
      ],
      "group_id": null,
      "description": null,
      "difficult": false,
      "shape_type": "rectangle",
      "flags": {},
      "attributes": {},
      "kie_linking": []
    },
    {
      "label": "figure",
      "score": 0.9148626327514648,
      "points": [
        [
          39.46824111938477,
          575.4597023010255
        ],
        [
          134.98223495483398,
          575.4597023010255
        ],
        [
          134.98223495483398,
          652.0944175720215
        ],
        [
          39.46824111938477,
          652.0944175720215
        ]
      ],
      "group_id": null,
      "description": null,
      "difficult": false,
      "shape_type": "rectangle",
      "flags": {},
      "attributes": {},
      "kie_linking": []
    },
    {
      "label": "figure",
      "score": 0.8906117677688599,
      "points": [
        [
          29.65818214416504,
          303.123551940918
        ],
        [
          119.6738712310791,
          303.123551940918
        ],
        [
          119.6738712310791,
          366.44710540771484
        ],
        [
          29.65818214416504,
          366.44710540771484
        ]
      ],
      "group_id": null,
      "description": null,
      "difficult": false,
      "shape_type": "rectangle",
      "flags": {},
      "attributes": {},
      "kie_linking": []
    },
    {
      "label": "figure",
      "score": 0.7079694271087646,
      "points": [
        [
          32.23932399749756,
          155.44689559936523
        ],
        [
          128.2761215209961,
          155.44689559936523
        ],
        [
          128.2761215209961,
          245.8070785522461
        ],
        [
          32.23932399749756,
          245.8070785522461
        ]
      ],
      "group_id": null,
      "description": null,
      "difficult": false,
      "shape_type": "rectangle",
      "flags": {},
      "attributes": {},
      "kie_linking": []
    }
  ],
  "imagePath": "1265030679.png",
  "imageData": null,
  "imageHeight": 1064,
  "imageWidth": 314,
  "description": ""
}

这是我自己用同一个模型对同一个数据推理出的bbox的信息
[
    {
        "confi": 0.72,
        "coordi": [
            30.26,
            426.85,
            116.82,
            521.59
        ]
    },
    {
        "confi": 0.53,
        "coordi": [
            0.47,
            94.39,
            138.78,
            149.19
        ]
    },
    {
        "confi": 0.4,
        "coordi": [
            29.91,
            171.58,
            129.15,
            396.58
        ]
    },
    {
        "confi": 0.37,
        "coordi": [
            3.5,
            570.09,
            138.45,
            679.09
        ]
    },
    {
        "confi": 0.31,
        "coordi": [
            0.0,
            569.82,
            192.4,
            677.7
        ]
    },
    {
        "confi": 0.3,
        "coordi": [
            0.38,
            948.43,
            119.61,
            979.7
        ]
    },
    {
        "confi": 0.29,
        "coordi": [
            29.73,
            241.23,
            128.19,
            395.48
        ]
    },
    {
        "confi": 0.27,
        "coordi": [
            0.0,
            0.0,
            128.52,
            24.73
        ]
    },
    {
        "confi": 0.26,
        "coordi": [
            1.71,
            101.7,
            194.85,
            144.4
        ]
    }
]

**Comment by VikingDeng** [comment]
以防万一我再给出更多的信息
这是x-anylabeling用我的模型推理出的bbox的信息
```json
{
“version”： “2.4.4”，
“flags”： {}，
“shapes”： [
{
“label”： “figure”，
“score”： 0.9150258302688599，
“points”： [
[
29.914649963378906，
428.7963348388672
]，
[
116.09727401733399，
428.7963348388672
]，
[
116.09727401733399， 510.57238006591797 ]， [ 29.914649963378906，
510.57238006591797
]
]，
“group_id”： null，
“description”： null，
“困难”： false，
“shape_type”： “矩形”，
“flags”： {}，
“attributes”： {}，
“kie_linking”： []
}，
{
“label”： “figure”，




“分数”： 0.9148626327514648，
“积分”： [
[
39.46824111938477，
575.4597023010255
]，
[
134.98223495483398，
575.4597023010255
]，
[
134.98223495483398，
652.0944175720215
]，
[
39.46824111938477，
652.0944175720215
]
]，
“group_id”：null，
“description”： null，
“difficult”： false，
“shape_type”： “矩形”，
“flags”： {}，
“attributes”： {}，
“kie_linking”： []
}，
{
“label”： “figure”，
“score”： 0.8906117677688599，
“points”： [
[
29.65818214416504，
303.123551940918
]，
[
119.6738712310791，
303.123551940918 ]， [ 119.6738712310791，
366.44710540771484
]，
[
29.65818214416504，
366.44710540771484
]
]，
“group_id”： null，
“description”： null，
“困难”： false，
“shape_type”： “矩形”，
“flags”： {}，
“属性”： {}，
“kie_linking”： []
}


，
{
“label”： “figure”，
“score”： 0.7079694271087646，
“points”： [
[
32.23932399749756，
155.44689559936523
]，
[
128.2761215209961，
155.44689559936523
]，
[
128.2761215209961，
245.8070785522461
]，
[
32.23932399749756，
245.8070785522461
]
]，
“group_id”： null，
“description”： null，
“difficult”： false，
“shape_type”： “rectangle”，
“flags”： {}，
“attributes”： {}，
“kie_linking”： []
}
]，
“imagePath”： “1265030679.png”，
“imageData”： null，
“imageHeight”： 1064，
”imageWidth“： 314，
”description“： ””
}
```
这是我自己用同一个模型对同一个数据推理出的bbox的信息
```json

[
{
“confi”： 0.72，
“coordi”： [
30.26，
426.85，
116.82，
521.59
]
}，
{
“confi”： 0.53，
“coordi”： [
0.47，
94.39，
138.78，
149.19
]
}，
{
“confi”： 0.4，
“coordi”： [
29.91，
171.58，
129.15，
396.58
]
}，
{
“confi”： 0.37，
“coordi”： [
3.5，
570.09，
138.45，
679.09
]
}，
{
“confi”： 0.31，
“coordi”： [
0.0，
569.82，
192.4，
677.7
]
}，
{
“confi”： 0.3，
“coordi”： [
0.38，
948.43，
119.61，
979.7
]
}，
{
“confi”： 0.29，
“coordi”： [
29.73，
241.23，
128.19，
395.48
]
}，
{
“confi”： 0.27，
“coordi”： [
0.0，
0.0，
128.52，
24.73
]
}，
{
“confi”： 0.26，
“coordi”： [
1.71，
101.7，
194.85，
144.4
]
}
]
```

**Comment by CVHub520** [comment]
@VikingDeng Hey there! 😄 

Thank you for reporting this issue. I appreciate you taking the time to search for similar reports before raising this one.

Regarding the X-AnyLabeling Component issue, I have a few suggestions that may help resolve the problem:

  1. Try running the sample code you provided. This approach may yield results closer to the desired outcome than the current implementation in the X-AnyLabeling component.

```python
from ultralytics import YOLO

# Load a model
model = YOLO("yolo11n.pt")  # pretrained YOLO11n model

# Run batched inference on a list of images
results = model(["image1.jpg", "image2.jpg"])  # return a list of Results objects

# Process results list
for result in results:
    boxes = result.boxes  # Boxes object for bounding box outputs
    masks = result.masks  # Masks object for segmentation masks outputs
    keypoints = result.keypoints  # Keypoints object for pose outputs
    probs = result.probs  # Probs object for classification outputs
    obb = result.obb  # Oriented boxes object for OBB outputs
    result.show()  # display to screen
    result.save(filename="result.jpg")  # save to disk
```

2. If the Ultralytics-based code still doesn't resolve the issue, you could try setting breakpoints in the [key](https://github.com/CVHub520/X-AnyLabeling/blob/main/anylabeling/services/auto_labeling/__base__/yolo.py#L303) areas of the X-AnyLabeling code, as you mentioned, to help identify the root cause of the problem.

3. Additionally, I would recommend reviewing the [Ultralytics documentation](https://docs.ultralytics.com/modes/predict/#key-features-of-predict-mode), as it provides detailed information on the various features and usage of their YOLO models, which may help you troubleshoot the issue further.

Please let me know if the Ultralytics-based approach or the breakpoint debugging helps resolve the problem. I'm happy to provide any further assistance as needed.

Best regards

**Comment by VikingDeng** [comment]
天啊为什么
您的代码确实是与x-anylabeling的输出结果一致
非常感谢您的回复
这解决了我的大麻烦
实在是想破脑袋了
